cmd/alertmanager/main.go

```diff
@@ -423,3 +433,10 @@ func mustHostname() string {
 	}
 	return hostname
 }
+
+func md5HashAsMetricValue(data []byte) float64 {
+	sum := md5.Sum(data)
+	var bytes []byte = sum[0:8]
+	bits := binary.LittleEndian.Uint64(bytes)
+	return math.Float64frombits(bits)
```

Comment:
This can result in a NaN which will cause problems downstream. I'd suggest limiting to say 48 bits and sorting it as a value rather than bits.